metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -344,6 +366,13 @@ case class ScalafixProvider(
       rules: List[String],
   ): Future[ScalafixEvaluation] = {
     val isScala3 = ScalaVersions.isScala3Version(scalaTarget.scalaVersion)
+    val isSource3 = scalaTarget.scalac.getOptions().contains("-Xsource:3")
+    val isScala3Dialect = isScala3 || isSource3
+    val canRemoveUnused = !isScala3 ||
+      // https://github.com/scala/scala3/pull/17835
+      Seq("3.0", "3.1", "3.2", "3.3")
```

Comment:
I took note on the scalafix PR to update that whenever this is backported to 3.3 LTS. We can't do it upfront as Scalafix has a similar check.